packages/contracts/contracts/minter-suite/Minters/one-off/ClaimMinter.sol

```diff
@@ -0,0 +1,452 @@
+     * The price for subsequent tokens is calculated as base price + (tokenId * priceIncrementInWei).
+     * This allows for incremental pricing.
+     * The base price must be set before users can claim tokens.
+     * @param basePriceInWei_ Base price in wei for token 0.
+     * @param priceIncrementInWei_ Price increment in wei.
+     */
+    function configurePricePerTokenInWei(
+        uint256 basePriceInWei_,
+        uint256 priceIncrementInWei_
+    ) external {
+        // CHECKS
+        AuthLib.onlyCoreAdminACL({
+            coreContract: coreContractAddress,
+            sender: msg.sender,
+            contract_: address(this),
+            selector: this.configurePricePerTokenInWei.selector
+        });
+
+        // EFFECTS
+        basePriceInWei = basePriceInWei_;
+        priceIncrementInWei = priceIncrementInWei_;
+    }
+
+    /**
+     * @notice Sets the timestamp when claiming can begin.
+     * Only callable by the core contract's Admin ACL.
+     * @dev This function sets the earliest time when users can claim tokens.
+     * The timestamp should be provided in Unix timestamp format (seconds since epoch).
+     * @param timestampStart_ Unix timestamp when claiming can begin.
+     */
+    function configureTimestampStart(uint256 timestampStart_) external {
+        // CHECKS
+        AuthLib.onlyCoreAdminACL({
+            coreContract: coreContractAddress,
+            sender: msg.sender,
+            contract_: address(this),
+            selector: this.configureTimestampStart.selector
+        });
+
+        // EFFECTS
+        timestampStart = timestampStart_;
+    }
+
+    /**
+     * @notice Pre-mints `amount` tokens to this contract
+     * for later claiming by collectors. Only callable by the core contract's Admin ACL.
+     * @dev This function mints tokens directly to this contract address, marking them
+     * as pre-minted but not yet claimed. The tokens remain in this contract's possession
+     * until claimed by users through the claimToken function. This allows for controlled
+     * distribution of tokens with arbitrary claiming order.
+     * @param amount Number of tokens to pre-mint.
+     */
+    function preMint(uint256 amount) external nonReentrant {
+        // CHECKS
+        AuthLib.onlyCoreAdminACL({
+            coreContract: coreContractAddress,
+            sender: msg.sender,
+            contract_: address(this),
+            selector: this.preMint.selector
+        });
+
+        MaxInvocationsLib.preMintChecks({
+            projectId: projectId,
+            coreContract: coreContractAddress
+        });
+
+        // EFFECTS
+        // Mint tokens to this contract
+        for (uint256 i = 0; i < amount; i++) {
+            uint256 tokenId = minterFilter.mint_joo({
+                to: address(this),
+                projectId: projectId,
+                coreContract: coreContractAddress,
+                sender: msg.sender
+            });
+
```

Comment:
nit: add check that tokenId % 1_000_000 is <= 500 (500 could be immutable configured in constructor)

Reply:
Added in https://github.com/ArtBlocks/artblocks-contracts/pull/1857/commits/5964d5db534b18268f440e99717194483b28e807